server/src/main/java/org/opensearch/bootstrap/BootstrapContext.java

```diff
@@ -32,14 +32,16 @@
 package org.opensearch.bootstrap;
 
 import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.annotation.PublicApi;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.env.Environment;
 
 /**
  * Context that is passed to every bootstrap check to make decisions on.
  *
- * @opensearch.internal
+ * @opensearch.api
  */
+@PublicApi(since = "1.0.0")
```

Comment:
Same here, I think Bootstrap classes can/should stay internal?

Reply:
Not really, we expose them in `Plugin` class and allow plugins to contribute these checks :

```
    public List<BootstrapCheck> getBootstrapChecks() {
        return Collections.emptyList();
    }

```